linux/ibus-keyman/src/main.c

```diff
@@ -91,6 +95,22 @@ start_component(void) {
 
   g_object_unref(component);
   km_service_get_default(NULL);  // initialise dbus service
+}
+
+static void
+start_component(void) {
+  g_message("Starting ibus-engine-keyman");
+
+  ibus_init();
+
+  bus = ibus_bus_new();
+
+  if (ibus_bus_is_connected(bus)) {
+    add_keyboards(bus, NULL);
+  } else {
+    g_message("Waiting for ibus-daemon to start up...");
+    g_signal_connect(bus, "connected", G_CALLBACK(add_keyboards), NULL);
```

Comment:
Is it necessary to disconnect this signal at any time, e.g. when shutting down?

Reply:
Good question.
Since we're passing `NULL` for the `user_data` parameter, we don't have to explicitly disconnect. The signal handlers are automatically disconnected when `bus` gets finalized. 
See [Memory management of signal handlers](https://developer-old.gnome.org/gobject/stable/gobject-Signals.html#signal-memory-management).